test/wallet-rpc-test.js

```diff
@@ -318,6 +318,44 @@ describe('Wallet RPC Methods', function() {
     });
   });
 
+  it('should rpc abortRescan', async () => {
+    assert.strictEqual(wdb.height, 104);
+    assert.strictEqual(wdb.height, node.chain.height);
+
+    // When rescanning is not triggered
+    const response = await wclient.execute('abortrescan');
+    assert.strictEqual('rescanning is not running; abort does not make sense', response);
+
+    const handler = (wallet, data, details) => {
+        if (details.height === 51) {
+          wclient.execute('abortrescan');
+        }
+    };
+
+    wdb.on('confirmed', handler);
+
+    await wclient.execute('rescan', [0]);
+
+    assert.strictEqual(wdb.height, 51);
```

Comment:
1. This is failing *sometimes* I think it should be replaced with  a `forValue` wait to ensure wdb and node are at the same height

Reply:
This solved the issue for me...

```diff
diff --git a/test/wallet-rpc-test.js b/test/wallet-rpc-test.js
index dada15986..55db104ff 100644
--- a/test/wallet-rpc-test.js
+++ b/test/wallet-rpc-test.js
@@ -385,13 +385,13 @@ describe('Wallet RPC Methods', function() {
 
     await wclient.execute('rescan', [0]);
 
-    assert.strictEqual(wdb.height, 51);
+    // Can not garuntee where rescan will stop due to async
+    assert(wdb.height > 50);
+    assert(wdb.height < 60);
     wdb.removeListener('confirmed', handler);
   });
 
   it('should not "rollback to the future"', async () => {
-    assert.strictEqual(wdb.height, 51);
-
     await assert.rejects(
       wclient.execute('rescan', [60]),
       {message: 'WDB: Cannot rollback to the future.'}
@@ -399,7 +399,6 @@ describe('Wallet RPC Methods', function() {
   });
 
   it('should rpc rescan and finish rescan after abort', async () => {
-    assert.strictEqual(wdb.height, 51);
     await wclient.execute('rescan', [40]);
     assert.strictEqual(wdb.height, 105);
     assert.strictEqual(wdb.height, node.chain.height);

```